channels/rdpsnd/server/rdpsnd_main.c

```diff
@@ -662,7 +666,8 @@ int rdpsnd_server_handle_messages(RdpsndServerContext *context)
 		Stream_SetPosition(s, 0);
 		if (priv->expectedBytes)
 		{
-			Stream_EnsureCapacity(s, priv->expectedBytes);
+			if (!Stream_EnsureCapacity(s, priv->expectedBytes))
+				return -1;
```

Comment:
according to the return value documentation above - shouldn't it return 0 on error ?